applications/StructuralMechanicsApplication/custom_elements/truss_elements/linear_truss_element_2D.cpp

```diff
@@ -644,6 +644,54 @@ void LinearTrussElement2D<TNNodes>::CalculateOnIntegrationPoints(
 /***********************************************************************************/
 /***********************************************************************************/
 
+template<SizeType TNNodes>
+void LinearTrussElement2D<TNNodes>::CalculateOnIntegrationPoints(
+    const Variable<Vector>& rVariable,
+    std::vector<Vector>& rOutput,
+    const ProcessInfo& rProcessInfo
+    )
+{
+    const auto& integration_points = IntegrationPoints(GetIntegrationMethod());
+    rOutput.resize(integration_points.size());
+
+    if (rVariable == PK2_STRESS_VECTOR) {
+        ConstitutiveLaw::Parameters cl_values(GetGeometry(), GetProperties(), rProcessInfo);
+        cl_values.GetOptions().Set(ConstitutiveLaw::COMPUTE_STRESS             , true);
+        cl_values.GetOptions().Set(ConstitutiveLaw::COMPUTE_CONSTITUTIVE_TENSOR, true);
```

Comment:
yes, if the stiffness (E) is not required, you can set this to `false`

Reply:
Actually, if I am not wrong, the stress could be retrieved by the `CalculateOnIntegrationPoints(AXIAL_FORCE, --, --)` and then divided by the `GetPRoperties()[CROSS_AREA]` right? I mean implementing the new `CalculateOnIntegrationPoints(PK2_STRESS)` and calling the `CalculateOnIntegrationPoints(AXIAL_FORCE, --, --)` inside

Reply:
Richard, you are absolutely right. 
Alejandro, thank you for the info on `COMPUTE_CONSTITUTIVE_TENSOR`. I also made it false for other `CalculateOnIntegrationPoints` functions and there are no any failed tests. 
You are right it is possible to re-use `CalculateOnIntegrationPoints(AXIAL_FORCE,)`; however, 1) I have added `TRUSS_PRESTRESS_PK2`, 2) I would like to keep theses functions separately because `CalculateOnIntegrationPoints(AXIAL_FORCE,)` could be changed in future, for example, PK2 could be replaced with other stress. 
Next, Aljandro please could you advice how to get the stress vector for TImoshenko beams? As far as I can see there are two areas, `CROSS_AREA` and `AREA_EFFECTIVE_Y`.

Reply:
The Timoshenko constitutive laws compute the axial force, moment and shear forces. To obtain the stress vector one should compose the bending and axial contributions as the normal stress and the shear stresses with respecto to the shear force and shear area.

The `CROSS_AREA `is the area of the cross section used in axial loading. The `AREA_EFFECTIVE_Y` is the reduced shear area for evaluating the shear forces.